plugins/events-backend-module-github/src/http/createGithubSignatureVerifier.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import {
+  RequestDetails,
+  RequestValidationContext,
+  RequestValidator,
+} from '@backstage/plugin-events-node';
+import { verify } from '@octokit/webhooks-methods';
+
+/**
+ * Verifies that the request received is the expected GitHub request
+ * by using a secret token configured at GitHub and here
+ * to verify the signature received with the `x-hub-signature-256` header.
+ *
+ * See https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks
+ * for more details.
+ *
+ * @param config - root config
+ * @public
+ */
+export function createGithubSignatureVerifier(
+  config: Config,
+): RequestValidator {
+  const secret = config.getOptionalString(
+    'events.modules.github.webhookSecret',
+  );
+  if (!secret) {
```

Comment:
Maybe log a warning? This may be unexpected. Or, is it so bad that we should throw an exception? At least in production - that could be an alternative; log in development and throw in prod, kinda like we behave for server-to-server token secrets.

Reply:
yeah. It certainly could qualify for throwing an exception. Even though I would rather change `getOptionalString` to `getString` then. This would mean if you add this as request validator you *must* configure a secret.

However, I wonder whether someone might want to have a difference between environments (dev, stage, prod, ...) without the need to change the code or handle it yourself at code level.

`githubWebhookEventsModule` would add it this, too. Of course, you could just register the `github` topic via the config without this module.

I lean a bit towards making it required 🤔